Separate rvc

nickhsmith opened the pull request.

Commit:
removed obsolete file

Commit:
changed project folder structure for initialisation

Commit:
API change: separated config and sample annotation parsers

Commit:
API change: separated config and sample annotation parsers

Commit:
fixed missin root path

Commit:
fixed missin root path

Commit:
import config differently

Commit:
import config differently

Commit:
package finding for subpackage

Commit:
package finding for subpackage

Commit:
added circleci config

Commit:
removed subworkflow and running pipeline for aberrantExpression; new functions introduced for Script-Rule-Html conversions; analysis script directory renamed; pipeline code copied to different destination

Commit:
removed subworkflow for AS and MAE; changed folder structure; using log for tmp snakemake objects

Commit:
fixed input flags for analysis scripts

Commit:
solved merge conflict

Commit:
set submodules to master

Commit:
Merge branch 'master' into pipeline_reruns

Commit:
applied branch-specific modifications

Commit:
removed submodules.py and refactored path setup

Commit:
fixed Readme and config copy errors

Commit:
fixed copying helpers functions

Commit:
using wbuild with fixed Readme.html functionality and saving wbuild config in DropConfig

Commit:
moved exportCounts into separate class

Commit:
parsing GENE_COUNT_FILE column in sample annotation

Commit:
removed wbuild from conda recipe

Commit:
using bioconda version of drop for dependencies, pip install drop, no conda building

Commit:
remove bioconda wbuild

Commit:
install wbuild with expl pip

Commit:
remove wbuild with pip

Commit:
get count info from AE class

Commit:
some refactoring and saving external counts IDs in separate dictionary

Commit:
removed ID renaming in merge script (naming already in counts script); fixed missing input bug in DropConfig

Commit:
add external counts to count files

Commit:
merge counts and coverage including external counts

Commit:
file checks for config and sample annotation reimplemented

Commit:
config file getters for submodules

Commit:
refactored export counts

Commit:
solved merge conflict

Commit:
travis run export counts rule

Commit:
Merge branch 'master' of github.com:gagneurlab/drop into import_counts

Commit:
reapplied bcftools command modification

Commit:
Merge branch 'master' of github.com:gagneurlab/drop into import_counts

Commit:
updated documentation to include count import

Commit:
splicing export counts all columns added

Commit:
resolved requested formality errors

Commit:
fixed requested MAE changes

Commit:
removed scanBamParam

Commit:
create missing columns for MAE results

Commit:
rnaVariantCalling update Scripts and config py files

Commit:
update drop demo caller

Commit:
update travis to run each module separately

Commit:
remove priority from rnaVariantCalling Snakefile

Commit:
remove memory values from java-options

Commit:
add messages to reorderSam

Commit:
wrapped picard in gatk

Commit:
correct usage of picard gatk wrapper

Commit:
create Sequence Dict

Commit:
update inputs for gatk4.1.8.1

Commit:
all tests travis

Commit:
added internal sampleID header check

Commit:
update mae genome arg, update cli template for RVC

Commit:
update documentation

Commit:
sort without bedtools

Commit:
update prepare doc

nickhsmith committed:
Update Snakefile

Commit:
remove temp() from sortBam input

Commit:
Merge branch 'rnaVariantCalling' of https://github.com/nickhsmith/drop into rnaVariantCalling

Commit:
add 30 sec latency to modules

Commit:
set travis script to only run RVC pipeline

Commit:
lengthen install time

Commit:
lengthen install time

Commit:
attempt to settle merge disputes

Commit:
change internal sample_annotation_relative.tsv to include RVC column

Commit:
fix module uploads, typos, and data paths

nickhsmith committed:
Merge branch 'dev' into dev

nickhsmith committed:
Update SampleAnnotation.R

remove extra '}'

nickhsmith committed:
Update Snakefile

remove duplicate install step

Commit:
temporary and poorly done hot-fix. Allows version installed to be >= version required (using string comparison)

Commit:
Merge branch 'dev' of https://github.com/nickhsmith/drop into dev

updating

nickhsmith committed:
Update installRPackages.R

undo version check

Commit:
add a final output to RVC pipeline

Commit:
modify bam_links_location

Commit:
update output touch

Commit:
Merge branch 'RVC_dev' of https://github.com/nickhsmith/drop into RVC_dev

pull to update

Commit:
add tests RVC

Commit:
add dependency graph RVC

Commit:
change rnaVariantCalling.ouput to rnaVariantCalling.input

Commit:
fix RVC tests

Commit:
fix RVC test

Commit:
fix RVC test

Commit:
use fread

Commit:
action storage RVC

testinguser committed:
Add changes

Commit:
test RVC

Commit:
Merge branch 'RVC_dev' of https://github.com/nickhsmith/drop into RVC_dev

pull

Commit:
test RVC with saving files

Commit:
test RVC with saving files

Commit:
test RVC with saving files

Commit:
test RVC with saving files

Commit:
shortened RVC tests. combined with git push

Commit:
test conditional

Commit:
test conditional

Commit:
test conditional

Commit:
test conditional

Commit:
test conditional

Commit:
test conditional

Commit:
test conditional

Commit:
test conditional

Commit:
test conditional

testinguser committed:
Add changes

Commit:
update tests to match GATK update (4.1.8.1 -> 4.1.9.0)

nickhsmith committed:
Merge branch 'dev' into RVC_dev

Commit:
update template config to include RVC add basic config/test_RVC.py

nickhsmith committed:
Merge pull request #1 from nickhsmith/RVC_dev

Rvc dev

Commit:
merge RVC with separate modules

Commit:
RVC pipeline

Commit:
rvc seperate modules

Commit:
update indexing command

Commit:
merge RVC with dev

Commit:
fix index function name

Commit:
update snakefile

Commit:
add norun test RVC

Commit:
remove ending spaces

Commit:
legacy handling and process SA without RVC column

Commit:
Merge branch 'dev' of https://github.com/gagneurlab/drop into separate_RVC

Commit:
add genome class and path finding to RVC

Commit:
demo config RVC run true

nickhsmith committed:
merge with dev

nickhsmith committed:
RVC control and BQSR fix

nickhsmith committed:
Merge remote-tracking branch 'origin/dev' into separate_RVC

nickhsmith committed:
make mask_file and known_site files match input data style

nickhsmith committed:
Merge remote-tracking branch 'origin/dev' into separate_RVC

nickhsmith committed:
update Snakefile and plotting

nickhsmith committed:
sampleParams and fix MAE bug

nickhsmith committed:
Update filterSNVs.sh

nickhsmith committed:
fix mae filter

nickhsmith committed:
Update filterSNVs.sh

add an if/else pattern. if QC don't filter, if not QC filter to heterozygous

nickhsmith committed:
update tests

nickhsmith committed:
update test

nickhsmith committed:
add comment and precision grep

nickhsmith committed:
hardcode UCSC results read in

nickhsmith committed:
update filter and results

nickhsmith committed:
update sampleAnnotation

nickhsmith committed:
update filter SNV

nickhsmith committed:
update filter SNV

nickhsmith committed:
standardize the MAE filter

nickhsmith committed:
update comment clarity

nickhsmith committed:
update comment clarity

nickhsmith committed:
fix test

nickhsmith committed:
update tests

nickhsmith committed:
update tests and base_config

nickhsmith committed:
make RVC:bqsr known_params always temp

nickhsmith committed:
cp tmp and tbi files

nickhsmith committed:
fix test_RVC

nickhsmith committed:
log to stdout to see errors

nickhsmith committed:
bqsr:add bai remove log

nickhsmith committed:
add full path to RVC

nickhsmith committed:
bqsr:add bai remove log

nickhsmith committed:
explicit GATK tmp-dir

nickhsmith committed:
will fail. ls tmpdir

nickhsmith committed:
change tmpdir

nickhsmith committed:
change tests

nickhsmith committed:
remove batch_0

nickhsmith committed:
batch0 only

Commit:
add sample specific variable names

Commit:
add batch_1. add full path to test

Commit:
full test

nickhsmith committed:
Merge branch 'dev' into mae_filter_fix

Commit:
remove snakemake hardcoded config

Commit:
update SampleParams

Commit:
fix sampleParams to include RVC

nickhsmith committed:
add helper functions

nickhsmith committed:
remove intermediate batch vcf and test

Commit:
add tbi as input

Commit:
change chr_conversion file loc. update RVC helpers

Commit:
minor edits

nickhsmith committed:
clean RVC snakefile

Commit:
github pat removed

nickhsmith committed:
add GITHUB_PAT

nickhsmith committed:
undo .github/workflows/GITHUB_PAT

nickhsmith committed:
Update python-package-conda.yml

nickhsmith committed:
unify reference_genome prep

nickhsmith committed:
Update requirementsR.txt

nickhsmith committed:
Update requirementsR.txt

devtools first

nickhsmith committed:
env: github_PAT

nickhsmith committed:
env: github_PAT for pytest

nickhsmith committed:
remove hardcoded config

nickhsmith committed:
update tMAE

nickhsmith committed:
Merge branch 'minor_updates' into separate_RVC

nickhsmith committed:
Merge branch 'dev' into separate_RVC

nickhsmith committed:
update docs

nickhsmith committed:
rename RVC demo_files

nickhsmith committed:
rename RVC demo_files

nickhsmith committed:
write newHeader

nickhsmith committed:
remove soft link for changeHeader

Commit:
hard code remove temp files

Commit:
undo copy of temp if file style matches

Commit:
update file

Commit:
undo RVC if not running

Commit:
force no vcf duplicates

Commit:
force no duplicates after splitting

nickhsmith committed:
Merge branch 'dev' into mae_filter_fix

mumichae committed:
Update .gitignore

mumichae committed:
Delete __init__.pyc

nickhsmith committed:
Merge branch 'dev' into separate_RVC

Commit:
update docs and filterSNV.sh

Commit:
Merge branch 'mae_filter_fix' of github.com:gagneurlab/drop into mae_filter_fix

Commit:
update filterSNV

nickhsmith committed:
remove old file

nickhsmith committed:
Update mergeCounts.R

Commit:
update comments

Commit:
indent

nickhsmith committed:
Merge branch 'dev' into separate_RVC

vyepez88 committed:
Update index.rst

vyepez88 committed:
Update installation.rst

vyepez88 committed:
Update pipeline.rst

vyepez88 committed:
Update prepare.rst

vyepez88 committed:
Update prepare.rst

vyepez88 commented on docs/source/prepare.rst:
```diff
@@ -123,17 +123,45 @@ padjCutoff             numeric    Same as in aberrant expression.
+
+repeat_mask            character  Location of the RepeatMask .bed file.                                                                                                                                  ``path/to/RepeatMask.bed``
+minAlt                 numeric    Integer describing the minimum required reads that support the alternative allele. We recommend using a minimum of 3 if further filtering on your own. 10 otherwise.   ``3``
+hcArgs                 character  String describing additional arguments for GATK haplocaller. For expert tuning.                                                                                        ``""``
```
Add the expert tuning to the Advanced section

vyepez88 committed:
Update config_relative.yaml

vyepez88 commented on drop/modules/mae-pipeline/MAE/Results.R:
```diff
@@ -41,9 +41,14 @@ suppressPackageStartupMessages({
 rmae <- lapply(snakemake@input$mae_res, function(m){
   rt <- readRDS(m)
+  # force consistant UCSC chromosome style
+  rt <- rt[!grepl("chr",contig),contig:= paste0("chr",contig)]
```
this we don't want

vyepez88 commented on drop/modules/rvc-pipeline/GATK_BASH/indexReads.sh:
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
```
we don't index anymore

vyepez88 commented on drop/template/Scripts/rnaVariantCalling/Overview.R:
```diff
@@ -0,0 +1,44 @@
+
+saveRDS(snakemake, snakemake@log$snakemake)
+
+#+ eval=TRUE, echo=FALSE
```
I think `sep` is not needed. You could simplify lines 33-39 into the following:
`plot_table <- lapply(snakemake@input$countFiles, fread) %>% rbindlist()`

vyepez88 commented on drop/template/Scripts/rnaVariantCalling/Overview.R:
```diff
@@ -0,0 +1,44 @@
+)
+
+plot_table <- rbindlist(plot_table)
+plot_table$GT <- as.factor(plot_table$GT)
```
use data.table commands `plot_table[, GT := as.factor(GT)]`

vyepez88 commented on drop/template/Scripts/rnaVariantCalling/Overview.R:
```diff
@@ -0,0 +1,44 @@
+
+plot_table <- rbindlist(plot_table)
+plot_table$GT <- as.factor(plot_table$GT)
+ggplot(melt(plot_table,measure.vars = 3:5),aes(x = variable, y= value,fill=GT)) + geom_boxplot() + 
```
Can you add a comment on what are you trying to plot? # of variants per sample?

vyepez88 commented on drop/template/Scripts/rnaVariantCalling/Overview.R:
```diff
@@ -0,0 +1,44 @@
+ggplot(melt(plot_table,measure.vars = 3:5),aes(x = variable, y= value,fill=GT)) + geom_boxplot() + 
+  ylab("Number of Variants") + xlab("")
+
+DT::datatable(plot_table, filter = 'top')
```
not sure how big this can be. Consider limiting the number of rows.

vyepez88 commented on drop/template/config.yaml:
```diff
@@ -6,18 +6,22 @@ indexWithFolderName: true # whether the root base name should be part of the ind
 sampleAnnotation: # path to sample annotation (see documenation on how to create it)
 
+root: Output
+sampleAnnotation: Data/sample_annotation.tsv
```
why do we have 2 sample annotation entries?

vyepez88 commented on drop/template/config.yaml:
```diff
@@ -6,18 +6,22 @@ indexWithFolderName: true # whether the root base name should be part of the ind
+genome: Data/chr21.fa
+
+hpoFile: null
+random_seed: true  # just for demo data, remove for proper analysis
```
I thought we wanted a specific seed for the demo

vyepez88 commented on drop/template/config.yaml:
```diff
@@ -6,18 +6,22 @@ indexWithFolderName: true # whether the root base name should be part of the ind
     geneAnnotations:
       - v29
-    excludeGroups:
+    excludeGroups
```
why remove the colon here and not in other instances?

Commit:
focus on batch_vcf.

Commit:
use DROP_GROUPS

Commit:
update tests and demo

Commit:
remove RNA_VARIANT_GROUP

Commit:
remove priority

Commit:
handle gtf.gz file

Commit:
improve 'grep' chromosome matching

Commit:
improve config tracking and file handling

Commit:
allow for config deletion

Commit:
update RVC minALT default

Commit:
update RVC  default

Commit:
update checkFiles utils

Commit:
update tmp resources

Commit:
update test paths

Commit:
markdown -> rmarkdown

Commit:
update RVC

Commit:
add frequency params

Commit:
add CountVariants

Commit:
add AF values

nickhsmith committed:
Dev (#292)

* Update README.md

* Update filterSNVs.sh

* Update filterSNVs.sh

add an if/else pattern. if QC don't filter, if not QC filter to heterozygous

* add comment and precision grep

* hardcode UCSC results read in

* standardize the MAE filter

* update comment clarity

* force no duplicates after splitting

* update docs and filterSNV.sh

* remove redundant conversion to UCSC style

* remove '.' from test

* add gene ranges

* ranges added

* ranges added

* fix tests for '.'

* Update .gitignore

* labels read count ratio fixed

* ranges fixed

* update authors

* Update Datasets.R

* fix qc plot

* Update DNA_RNA_matrix_plot.R

* hpo function bug fixed

* mae links and populations fixed

* hpo function fixed

* mae genes bug fix

* Update Results.R

* Update Results.R

* speed up hpo function

* Update Results.R

* rmarkdown instead of markdown

* Update Results.R

* Update test_MAE.py

* update conda to mamba install (#289)

* update conda to mamba install

* bump tMAE

Co-authored-by: Smith Nicholas <smith@in.tum.de>

* Update requirementsR.txt

* Update README.md

* Update setup.py

Bump 1.1.2

* Update setup.cfg

Bump 1.1.2

* Update conf.py

* Update __init__.py

Bump 1.1.2

* Update cli.py

Bump 1.1.2

Co-authored-by: Vicente Yepez <30469316+vyepez88@users.noreply.github.com>
Co-authored-by: Smith Nicholas <smith@in.tum.de>
Co-authored-by: Vicente <yepez@in.tum.de>

Commit:
merge with master

Commit:
update for tMAE

Commit:
update MAFDB download

nickhsmith committed:
cleanup download and MAE

nickhsmith committed:
remove MAE.gnomAD unused key

Commit:
remove keys object for exportCounts

Commit:
Merge branch 'separate_RVC' of github.com:gagneurlab/drop into separate_RVC

Commit:
Merge branch 'dev' into separate_RVC

Commit:
update mergeParams to include external count info

Commit:
fix import counts group matching

Commit:
Merge branch 'update_AE_mergeParam' into separate_RVC
